src/tennisSimulator/Set.java

```diff
@@ -0,0 +1,70 @@
+package tennisSimulator;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Set {
+
+    private final Player firstPlayer;
+    private final Player secondPlayer;
+    private int setCounter;
+    private final int numberOfGamesInSet;
+    private final Map<Integer, String> setResults;
+
+    public Set(Player firstPlayer, Player secondPlayer, int numberOfGamesInSet) {
+        this.firstPlayer = firstPlayer;
+        this.secondPlayer = secondPlayer;
+        setCounter = 1;
+        this.numberOfGamesInSet = numberOfGamesInSet;
+        setResults = new HashMap<>();
+    }
+
+    public void saveResult(int firstPlayerGamesWon, int secondPLayerGamesWon) {
+        String setResult = firstPlayerGamesWon + "-" + secondPLayerGamesWon;
+        setResults.put(setCounter, setResult);
+        setCounter++;
+    }
+
+    public Map<Integer, String> getSetResults() {
+        return setResults;
+    }
+
+    public void resetResult() {
+        secondPlayer.getMatchStats().resetGamesWon();
+        firstPlayer.getMatchStats().resetGamesWon();
+    }
+
+    public boolean hasTieBreak() {
+        return firstPlayer.getMatchStats().getGamesWon() == numberOfGamesInSet
+                && secondPlayer.getMatchStats().getGamesWon() == numberOfGamesInSet;
+    }
+
+    public Player playerWhichWonSet() {
+        if (firstPlayerHasWonSet()) {
+            return firstPlayer;
+        } else if (secondPlayerHasWonSet()) {
+            return secondPlayer;
+        }
+        return null;
+    }
+
+    private boolean firstPlayerHasWonSet() {
```

Comment:
This one and the method below are the same just with different player instances. I wonder maybe player class can tell if the player has won set. Because now it looks like duplicate code

Reply:
Maybe this also should be moved into separate task. I believe same problem with "train-wreck" anti pattern will exist. So at first responsibilities of each class must be reviewed.

Reply:
Agree